app/controllers/sog/name-trainer.js

```diff
@@ -0,0 +1,145 @@
+import Controller from '@ember/controller';
+import { computed } from '@ember/object';
+import GroupMembershipsMixin from 'alpha-amber/mixins/group-memberships-mixin';
+import { run } from '@ember/runloop';
+
+export default Controller.extend(GroupMembershipsMixin, {
+  queryParams: ['group', 'difficulty'],
+  group: null,
+  difficulty: 1,
+  selectedGroup: null,
+  difficultyOptions: [
+    {
+      value: 1,
+      label: 'Makkelijk'
+    }, {
+      value: 2,
+      label: 'Gemiddeld'
+    }, {
+      value: 3,
+      label: 'Moeilijk'
+    }
+  ],
+  current: 1,
+  success: 0,
+  questions: [],
+  textInput: '',
+  started: false,
+  finished: false,
+  answered: false,
+  groupSelected: computed('group', 'model', function() {
+    this.generateQuestions(this.get('model'));
+    return this.get('group') != null;
+  }),
+  humanizedDifficulty: computed('difficultyOptions', 'difficulty', function() {
+    return this.get('difficultyOptions').find(option => option.value === this.get('difficulty')).label;
+  }),
+  currentQuestion: computed('current', 'questions', function() {
+    return this.get('questions').objectAt(this.get('current') - 1).question;
+  }),
+  currentOptions: computed('current', 'questions', function() {
+    return this.get('questions').objectAt(this.get('current') - 1).options;
+  }),
+  currentQuestionItem: computed('current', 'questions', function() {
+    return this.get('questions').objectAt(this.get('current') - 1);
+  }),
+  progress: computed('current', 'questions.length', function() {
+    return (100 / this.get('questions.length')) * this.get('current');
+  }),
+  inputClass: computed('currentQuestionItem', 'answered', function() {
+    if (this.get('answered')) {
+      if (this.get('currentQuestionItem').success) {
+        return 'form-control is-valid';
+      } else {
+        return 'form-control is-invalid';
+      }
+    }
+
+    return 'form-control';
+  }),
+  grade: computed('questions.length', 'success', function() {
+    return 1 + Math.round((90 / this.get('questions.length')) * this.get('success')) / 10;
+  }),
+  actions: {
+    startTrainer() {
+      if (this.get('group') == null) {
+        this.set('group', this.get('selectedGroup.id'));
```

Comment:
Waarom zijn dit 2 losse entiteiten? Kan je niet direct group aanpassen met de dropdown?

Reply:
Nou, dit is met opzet om 2 redenen:
1. Je kunt nu ook een link delen naar een specifieke vragenlijst
2. Nu gaat hij pas gebruikers laden zodra je echt met de quiz start, en niet daarvoor al.